src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -629,17 +629,23 @@ where
         Ok(pkg.clone())
     };
 
-    fn multi_err(kind: &str, mut pkgs: Vec<&Package>) -> String {
+    fn multi_err<T>(kind: &str, source: &T, mut pkgs: Vec<&Package>) -> String
+    where
+        T: Source,
```

Comment:
What about this, given we already assume it is from a git repo? Then we can avoid generic and make it more explicit.

```suggestion
    fn git_multi_pkgs_error<T>(kind: &str, git_url: &str, mut pkgs: Vec<&Package>) -> String
```
```